### ISSUE TITLE
How can eg. `npx` execute `yarn`

### ISSUE THREAD

**Issue by vladdeSV** [teaxyz/cli#381]
I don't have node, npm, npx, or yarn traditionally installed, only in `~/.tea`.

With tea magic, if I run the command `npx browserslist@latest --update-db` in a repo using yarn, I get the following error:

```
$ npx browserslist@latest --update-db
/bin/sh: line 1: yarn: command not found
node:child_process:960
    throw err;
    ^

Error: Command failed: yarn npm info caniuse-lite --json
/bin/sh: line 1: yarn: command not found
```

My guess is that tea magic doesn't work if another process calls software that is not installed. What is an appropriate solution for this?

**Comment by mxcl** [comment]
what happens on a non-tea system if a user doesn’t have `yarn` installed?

**Comment by mxcl** [comment]
So to get other pkgs into the env you can do this:

```sh
tea +yarnpkg.com npx browserslist@latest --update-db
```

which is tedious. My thinking for this sort of situation going forward is to, where there is magic and, where we can, add command not found handling.

But we first need to catalog scenarios where we *should*. This may be one of them, but my question above piques curiosity still… ie. what would a user do on a non tea system?

**Comment by vladdeSV** [comment]
**tl;dr**: using `tea +yarnpkg.com +npmjs.com npx ...` worked. the whole thing seems like a bug with the packaged yarn version.

---

> what happens on a non-tea system if a user doesn’t have yarn installed?

From what I have tested, browserlist checks which type of lockfile exists in the project.

(In this I have node and npm installed)

w/o yarn + yarn.lock:
```
issue-381 $ ls -a
package.json  yarn.lock

issue-381 $ npx browserslist@latest --update-db
/bin/sh: line 1: yarn: command not found
node:child_process:960
    throw err;
    ^
```

w/o yarn + package-lock.json:
```
$ ls -a
package-lock.json  package.json

$ npx browserslist@latest --update-db
Latest version:     1.0.30001452
Installed version:  none
// snip
No target browser changes
```

---

> So to get other pkgs into the env you can do this:

I had to add `+npmjs.com` and it worked :tada:

```
tea +yarnpkg.com +npmjs.com npx browserslist@latest --update-db
```

---

Finally, and this is unrelated to the issue, but tea's version of yarn is the `-dev` variant, and give a weird message whenever yarn is run:

```
$ which yarn
which: no yarn in (...)

$ yarn --version
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
3.4.1-dev 
```

I did not understand this when I posted my issue, but this seems to be a bug with this version of yarn - I should not need to update caniuse-lite.

**Comment by vladdeSV** [comment]
I am not sure if the terminology is right in "child process calling non-installed", but I can provide another example where some software assumes something is installed:

For Rust, using the VS Code extension rust-analyzer, it errors when it tried calling `rustfmt`

```
[Error - 2:31:33 PM] Request textDocument/formatting failed.
  Message: Failed to spawn "rustfmt" "--edition" "2021"
  Code: -32603 
```

If I call this from my terminal, it works as expected

```
$ echo "fn  main (){ }" | rustfmt --edition 2021
fn main() {}
```

**Comment by vladdeSV** [comment]
Ah, of course, if I start VS Code from my terminal, the extension can format my code.

```sh
tea +rust-lang.org code
```

It doesn't work perfectly, but right now I am unsure if the problem is with tea, the extension, or me.

**Comment by jhheider** [comment]
You can also use the symlinking feature for things like this; `sudo ln -sf $(which tea) /usr/local/bin/rustfmt`, etc. As an example, that's how I use `fishshell.com` as a login shell.

**Comment by mxcl** [comment]
So for now I think the solution is a `dev-env` where you add each needed pkg to the env. You can use `tea.yml` or add them as a `tea` node to `package.json`, or other options.

```json
"tea": {
    "dependencies": { "foo": "*" }
}
```

Longer term we may investigate adding our magic to things like `npx` though that is potentially too far. Needs thought.

**Comment by mxcl** [comment]
tea now adds everything installed to `~/.tea/.local/bin` and adds that to the PATH with magic so this will now work
